x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -332,10 +332,14 @@ private PhysicalOperation planTopN(TopNExec topNExec, LocalExecutionPlannerConte
                 throw new UnsupportedOperationException();
             }
 
+            TopNOperator.Encoder encoder = a.dataType() != DataTypes.IP
```

Comment:
@ChrisHegarty I haven't added a metadata element to blocks to indicate they are variable/fixed in length, but went for the "encoder" approach based on the field's data type at planning stage.

Reply:
I think we should be pretty explicit here about which types support which encoders. `KEYWORD` and `TEXT` get the utf-8 one, IP gets the other, and everything else gets null/nothing. But we should list each type we support. That way when we add a new type it isn't supported as a sort key until someone looks here and makes a choice. If it's fixed length they can go one way. If it's utf-8 they can go another. If it's some other type, well, they may need to write a new encoder.